# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../attr/0002-cdefs.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- a/include/xattr.h
+++ b/include/xattr.h
@@ -30,8 +30,13 @@
 #define XATTR_CREATE  0x1       /* set value, fail if attr already exists */
 #define XATTR_REPLACE 0x2       /* set value, fail if attr does not exist */
 
+#ifndef __THROW
+#define __THROW
+#endif
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 extern int setxattr (const char *__path, const char *__name,
 		      const void *__value, size_t __size, int __flags) __THROW;
@@ -58,6 +63,8 @@
 extern int lremovexattr (const char *__path, const char *__name) __THROW;
 extern int fremovexattr (int __filedes,   const char *__name) __THROW;
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* __XATTR_H__ */
